core/src/services/sled/backend.rs

```diff
@@ -145,7 +147,13 @@ impl kv::Adapter for Adapter {
     }
 
     async fn get(&self, path: &str) -> Result<Option<Vec<u8>>> {
-        self.blocking_get(path)
+        let cloned_self = self.clone();
+        let cloned_path = path.to_string();
+
+        task::spawn_blocking(move || cloned_self.blocking_get(cloned_path.as_str()))
+            .await
+            .map_err(new_task_join_error)
```

Comment:
Please use map_err(abc)? to return the error instead of and_than

Reply:
Got it.